GooglePlayInstant/Editor/PlayInstantQuickDeployInputManager.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.IO;
+using UnityEngine;
+
+namespace GooglePlayInstant.Editor
+{
+    /// <summary>
+    /// Provides functionality for reading, storing and retrieving text field values from "Deploy Bundle" section.
+    /// </summary>
+    public static class DeployBundleTextFields
+    {
+        private const string dataStorageFilePath = ".google_play_instant_quick_deploy_inputs.json";
+
+        private static string _oauth2CredentialsPath;
+        private static string _localAssetBundlePath;
+        private static string _remoteBucketName;
+        private static string _remoteObjectName;
+
+        public static string Oauth2CredentialsPath
+        {
+            get { return getUpdatedValue(ref _oauth2CredentialsPath); }
+            set { updateNewValue(ref _oauth2CredentialsPath, value); }
+        }
+
+        public static string LocalAssetBundlePath
+        {
+            get { return getUpdatedValue(ref _localAssetBundlePath); }
+            set { updateNewValue(ref _localAssetBundlePath, value); }
+        }
+
+        public static string RemoteBucketName
+        {
+            get { return getUpdatedValue(ref _remoteBucketName); }
+            set { updateNewValue(ref _remoteBucketName, value); }
+        }
+
+        public static string RemoteObjectName
+        {
+            get { return getUpdatedValue(ref _remoteObjectName); }
+            set { updateNewValue(ref _remoteObjectName, value); }
+        }
+
+
+        private static string getUpdatedValue(ref string field)
+        {
+            if (string.Equals(field, null))
+            {
+                RetrieveTextFieldInputs();
+            }
+
+            return field;
+        }
+
+        // Do nothing if value has not changed, otherwise update the value of the static field and write to file.
+        private static void updateNewValue(ref string field, string value)
+        {
+            if (string.Equals(field, value))
+            {
+                return;
+            }
+
+            field = value;
+            StoreTextFieldInputs();
+        }
+
+        private static void StoreTextFieldInputs()
```

Comment:
Did you consider EditorPrefs.SetString() for storing these?

There are pros/cons, but see https://github.com/google/play-instant-unity-plugin/blob/master/GooglePlayInstant/Editor/PlayInstantBuildConfiguration.cs for an example.

Reply:
Good to know. Changed my implementation. Better this way because I get to do a single write for just one key without having to rewrite the entire file